sw/device/lib/aes.c

```diff
@@ -0,0 +1,100 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/aes.h"
+
+#include "aes_regs.h"  // Generated.
+#include "sw/device/lib/common.h"
+
+#define AES0_BASE_ADDR 0x40110000
+
+void aes_init(aes_cfg_t aes_cfg) {
+  // Convert key_len from bytes to onehot
+  size_t key_len_onehot = 0b001;
+  if (aes_cfg.key_len_in_bytes == 24) {
+    key_len_onehot = 0b010;
+  } else if (aes_cfg.key_len_in_bytes == 32) {
+    key_len_onehot = 0b100;
+  }
+
+  REG32(AES_CTRL(0)) =
+      aes_cfg.mode << AES_CTRL_MODE |
+      (key_len_onehot & AES_CTRL_KEY_LEN_MASK) << AES_CTRL_KEY_LEN_OFFSET |
+      aes_cfg.manual_start_trigger << AES_CTRL_MANUAL_START_TRIGGER |
+      aes_cfg.force_data_overwrite << AES_CTRL_FORCE_DATA_OVERWRITE;
+};
+
+void aes_key_put(const void *key, size_t key_len_in_bytes) {
+  const size_t key_len_in_words = key_len_in_bytes >> 2;
+
+  for (int i = 0; i < key_len_in_words; ++i) {
+    REG32(AES_KEY0(0) + i * sizeof(uint32_t)) = ((uint32_t *)key)[i];
```

Comment:
I think your example would not work as such, it would have to be:
```
// description of what loop does
for (int i = 0; i < key_len_in_words; ++i) {
  uint32_t key_offset = i * sizeof(uint32_t);
  REG32(AES_KEY0(0) + key_offset) = ((uint32_t *)key)[i];
}
```
Personally, I don't think it really helps to make the code simpler but it adds yet another variable and one more line of code in a very simple loop. But I don't have a very strong opinion on this.

I definitely will add a description for the loop.